### PR TITLE
Integration tests for hdbview against postgresql and mysql 

### PR DESCRIPTION
**Pull request by Wildflowerr**
Fixes https://github.com/SAP/xsk/issues/168 when https://github.com/eclipse/dirigible/pull/860 is merged and released.

### CODE REVIEW

File: modules/engines/engine-hdb/pom.xml [hunk 1]
```diff
@@ -204,6 +204,12 @@
       <artifactId>mysql-connector-java</artifactId>
       <version>8.0.24</version>
     </dependency>
+    <dependency>
+      <groupId>com.sap.db.jdbc</groupId>
+      <artifactId>ngdbc</artifactId>
+      <version>2.4.63</version>
+    </dependency>
```
Comment (ThuF) [inline modules/engines/engine-hdb/pom.xml:211]: Add `test` scope?

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import static org.junit.Assert.assertTrue;
+
+public class XSKHDBViewParserHanaCase {
+
+  private static Connection connection;
+  private static IXSKHDBCoreFacade facade;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws SQLException, IOException {
+    JDBCModel model = new JDBCModel(HANA_DRIVER, Configuration.get("db.url"), Configuration.get("db.username"),
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java:48]: Extract `db.url`, `db.username` and `db.password` into constants, also the properties could be database specific e.g. `hana.url`, `hana.username` and `hana.password`.

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import static com.sap.xsk.hdb.ds.test.itest.utils.TestConstants.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class XSKHDBViewParserHanaCase {
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java:41]: Shouldn't all test classes ends with `test`: `XSKHDBViewParserHanaCase` -> `XSKHDBViewParserHanaTest`?

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    Injector injector = Guice.createInjector(new XSKHDBTestModule(model));
+    connection = injector.getInstance(DataSource.class).getConnection();
+    facade = injector.getInstance(Key.get(IXSKHDBCoreFacade.class, Names.named("xskHDBCoreFacade")));
+  }
+
+  @After
+  public void cleanUpAfterTest() throws SQLException {
+    Statement stmt = connection.createStatement();
+    stmt.executeUpdate(HDBVIEW_HANA_DROP_TABLE1_SQL);
+    stmt.executeUpdate(HDBVIEW_HANA_DROP_MY_VIEW1_SQL);
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java:59]: Shouldn't the `HDBVIEW_HANA_DROP_TABLE1_SQL` and `HDBVIEW_HANA_DROP_MY_VIEW1_SQL` constraints be part of the `XSKHDBViewParserHanaCase` test class?

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  @Before
+  public void setUpBeforeTest() throws SQLException {
+    Statement stmt = connection.createStatement();
+    stmt.executeUpdate(HDBVIEW_HANA_DROP_XSK_DATASTRUCTURES);
+    stmt.executeUpdate(HDBVIEW_HANA_CREATE_TABLE1_SQL);
+    stmt.executeUpdate(HDBVIEW_HANA_CREATE_MY_VIEW1_SQL);
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java:67]: Same about the constants

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  public static void setUpBeforeClass() throws SQLException, IOException {
+    JDBCModel model = new JDBCModel(HANA_DRIVER, Configuration.get("db.url"), Configuration.get("db.username"),
+        Configuration.get("db.password"));
+    Injector injector = Guice.createInjector(new XSKHDBTestModule(model));
+    connection = injector.getInstance(DataSource.class).getConnection();
+    facade = injector.getInstance(Key.get(IXSKHDBCoreFacade.class, Names.named("xskHDBCoreFacade")));
+  }
+
+  @After
+  public void cleanUpAfterTest() throws SQLException {
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java:56]: I think the order of the `@After` and `@Before` methods is changed.

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/utils/TestConstants.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2019-2021 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.hdb.ds.test.itest.utils;
+
+import org.eclipse.dirigible.commons.config.Configuration;
+
+public class TestConstants {
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/utils/TestConstants.java:16]: It's good idea to have such class, but it should contain only constants that are reused inside multiple test classes. Let's put the test specific constants inside the test class. In the end maybe only generic constants like `HANA_DRIVER`, `HANA_URL`, `HANA_USERNAME`, `HANA_PASSWORD`, etc. should be left here.

File: modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleHANAXSClassicView.hdbview [hunk 8]
```diff
@@ -0,0 +1,6 @@
+//schema name to be considered when itests are run on local
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleHANAXSClassicView.hdbview:1]: In the file location, do we need this `registry.public...` part? How about the following `registry.public.hdbview-itest/SampleHANAXSClassicView.hdbview` -> `hdbview-itest/SampleHANAXSClassicView.hdbview`?

File: modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleMySQLXSClassicView.hdbview [hunk 9]
```diff
@@ -0,0 +1,5 @@
+schema="test";
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleMySQLXSClassicView.hdbview:1]: Same as above

File: modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SamplePostgreXSClassicView.hdbview [hunk 10]
```diff
@@ -0,0 +1,5 @@
+schema="public";
```
Comment (ThuF) [inline modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SamplePostgreXSClassicView.hdbview:1]: Same as above

**Review by ThuF — CHANGES_REQUESTED** [review]

File: modules/engines/engine-hdb/pom.xml (hunk 1 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/pom.xml:212]: Done

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java (hunk 2 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java]: Done

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java (hunk 3 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java:41]: All itests end in ITCase(coming from integration testing case) except for the hana itests. This results from the need to provide parameters when we do run hana itests. That's why the pattern is changed from **/*ITCase to **/*HanaCase.
If this is not wanted, we can have a single itest profile with parameters serving all tests. I'm open to any naming of the tests if ITCase is not good enough.

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java (hunk 6 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java]: Removed @After clause. Create/Drop statements for additional tables is now moved in the test itself.

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java (hunk 4 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java]: Done as stated above.

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java (hunk 5 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java]: Constants are now removed except for the HANA ones.

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/utils/TestConstants.java (hunk 7 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/utils/TestConstants.java:16]: Test specific constants are now removed except for the hana ones.

File: modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleHANAXSClassicView.hdbview (hunk 8 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleHANAXSClassicView.hdbview]: Done.

File: modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleMySQLXSClassicView.hdbview (hunk 9 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SampleMySQLXSClassicView.hdbview]: Done

File: modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SamplePostgreXSClassicView.hdbview (hunk 10 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/resources/registry.public.hdbview-itest/SamplePostgreXSClassicView.hdbview]: Done

File: modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java (hunk 3 above)
Comment (Wildflowerr) [reply modules/engines/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/test/itest/hdbview/XSKHDBViewParserHanaCase.java]: As agreed all itests are now ending in ITTest pattern and there is a single itest profile.

**Review by ThuF — APPROVED** [review]
+1

**Pull request merged**